### PR TITLE
ci: Create job summaries in python

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: CI/physmon/phys_perf_mon.sh [hunk 1]
```diff
@@ -247,7 +247,10 @@ if [[ "$mode" == "all" || "$mode" == "simulation" ]]; then
     simulation geant4
 fi
 
-CI/physmon/summary.py $outdir/*.html $outdir/summary.html
+CI/physmon/summary.py $outdir/*.html summary.md
```
Comment (andiwand) [inline CI/physmon/phys_perf_mon.sh:250]: maybe the python script could output both `md` and `html` just to remove a bit of logic from bash

Comment (andiwand) [reply CI/physmon/phys_perf_mon.sh:250]: yeah I think we should preserve single source. so in python we could copy paste to the other output and prepend the html stuff

**Comment by paulgessinger** [comment]
The texme header doesn't work as well as I thought it would, and also the links don't currently work in the job summary (I have an idea how to fix that) so I'll switch to just using multiple separate outputs from python

**Review by andiwand — DISMISSED** [review]
nice!

**Comment by paulgessinger** [comment]
Ok, I added some functionality to herald to redirect to an artifact from a run id while maintaining paths. Now the physmon summary markdown should contain working links.

**Review by andiwand — DISMISSED** [review]

File: CI/physmon/summary.py [hunk 2]
```diff
@@ -34,29 +38,51 @@
         except Exception as e:
             print(r"could not parse {h}", e)
 
-with open(args.output, mode="w", encoding="utf-8") as f:
-    f.write(
-        """<!DOCTYPE html>
+output = "# physmon summary\n"
```
Comment (andiwand) [inline CI/physmon/summary.py:41]:
```suggestion
```
dead variable?

File: CI/physmon/summary.py [hunk 3]
```diff
@@ -34,29 +38,51 @@
...
+            """
         )
 
-    f.write(
-        """
-  </ul>
-</body>
-</html>"""
-    )
+is_ci = "GITHUB_ACTIONS" in os.environ
```
Comment (andiwand) [inline CI/physmon/summary.py:73]:
```suggestion
```
should we move this up?

File: CI/physmon/summary.py [hunk 4]
```diff
@@ -6,18 +6,22 @@
 import functools
 import os
 
+HERALD_URL = "https://herald.dokku.paulgessinger.com/view/{repo}/runs/{run_id}/artifacts/{artifact_name}/{path}"
```
Comment (andiwand) [inline CI/physmon/summary.py:9]:
```suggestion
HERALD_URL = "https://herald.dokku.paulgessinger.com/view/{repo}/runs/{run_id}/artifacts/{artifact_name}/{path}"
IS_CI = "GITHUB_ACTIONS" in os.environ
```

File: CI/physmon/summary.py [hunk 5]
```diff
@@ -34,29 +38,51 @@
...
-</body>
-</html>"""
-    )
+is_ci = "GITHUB_ACTIONS" in os.environ
+if args.md:
+    with open(args.md, mode="w", encoding="utf-8") as f:
+        f.write("# physmon summary\n")
+        for h, s in summary.items():
+            path = os.path.relpath(h, args.base)
+            if is_ci:
```
Comment (andiwand) [inline CI/physmon/summary.py:79]:
```suggestion
            if IS_CI:
```

**Review by andiwand — APPROVED** [review]

**Review by andiwand — APPROVED** [review]

**Pull request merged**